### PR TITLE
Quiz for Complete Recursion

### PR DESCRIPTION
**Pull request by dutinmeow**

### CODE REVIEW

**Comment by SansPapyrus683** [comment]
wwwait
where's deployment

**Comment by maggie-j-liu** [comment]
image

**Comment by SansPapyrus683** [comment]
image
soooooo did i get this right?

**Comment by SansPapyrus683** [comment]
ok so i didn't
the question is worded kinda poorly because i thought it was just iterating through the permutations and stuff

image
arrays are kinda niche, use a vector instead

**Comment by SansPapyrus683** [comment]
don't expect people in bronze to know about arrays
c arrays in general are just a pain to work with- they're error prone and hard to debug
just use vectors lol

**Comment by maggie-j-liu** [comment]
wait i'm so confused i thought people learned arrays before vectors??

**Comment by SansPapyrus683** [comment]
who the hell would learn arrays before vectors

**Comment by ryanchou-dev** [comment]
> image
> soooooo did i get this right?

yup, definitely reword this

**Comment by ryanchou-dev** [comment]
> I tried to make code as general as possible (for java and python users). Arrays are mostly similar across different languages, whereas vectors are completely different (ArrayList in Java vs Lists in Python)

you can do different code for diff languages right? <languagesection>

**Comment by dutinmeow** [comment]
I think next_permutation only exists in C++ and it defeats the point of the question to generate it with recursion. ~~also i suck with java and python~~. I can write different solutions for the third one tho.

**Comment by ryanchou-dev** [comment]
> I think next_permutation only exists in C++ and it defeats the point of the question to generate it with recursion. ~also i suck with java and python~. I can write different solutions for the third one tho.

i can help with python! and nathan would be good for java :D

**Comment by dutinmeow** [comment]
I mean I can do them but idk how to do next_permutation in other languages for the second question

**Comment by ryanchou-dev** [comment]
> I think next_permutation only exists in C++ and it defeats the point of the question to generate it with recursion. ~also i suck with java and python~. I can write different solutions for the third one tho.



> I think next_permutation only exists in C++

err then i think you shouldn't use it as a question, cuz it'll be irrelevant for python and java users.

**Comment by dutinmeow** [comment]
But the module itself mentions next_permutation. Also majority of CP people use c++

**Comment by ryanchou-dev** [comment]
> But the module itself mentions next_permutation. Also majority of CP people use c++

for other languages (since you can change questions for different lang settings)

**Comment by dutinmeow** [comment]
ok, ill change it. Can you guys write the other languages then? @ryanchou-dev @thecodingwizard

**Comment by nathangong** [comment]
What happened to the question that involves coding in the python and java sections? Are we not going to include them or are they waiting to be translated into the respective languages?

**Comment by dutinmeow** [comment]
I think we decide that we'er not going to include them

**Comment by ryanchou-dev** [comment]
> What happened to the question that involves coding in the python and java sections? Are we not going to include them or are they waiting to be translated into the respective languages?

yeah i think it was too inconsistent across languages, and the `next_permutation` error was more targeted for cpp users. (it wouldn't be necessary for python and java)

**Comment by caoash** [comment]
I learned arrays before vectors

**Comment by Aadit-Ambadkar** [comment]
> who the hell would learn arrays before vectors

Language

File: content/2_Bronze/Complete_Rec.mdx [hunk 1]
```diff
@@ -985,3 +990,324 @@ appear much less frequently then they once did.
...
+		It's definitely possible to run a $2^n$ solution for $n \leq 10$, but you might consider easier methods such as solutions that generate all permutations
+	      </Quiz.Explanation>
+	    </Quiz.Answer>
+	  </Quiz.Question>
+	  <Quiz.Question>
+	    Why are iterative methods of generating permutations and subsets like $\texttt{next\_permutation}$ and bitmasking preferable over recursive implementations?
+	    <Quiz.Answer>
+	      Recursion often runs into the issue of stack size, which can cause unexpected runtime errors
+	      <Quiz.Explanation>
+		A hallmark of problems featuring complete search with permutations/subsets is that $n$ is small, usually $\leq 20$. This means that recursion won't go particularly deep, and there won't be any issues with stack size.
```
Comment (Aadit-Ambadkar) [inline content/2_Bronze/Complete_Rec.mdx:1056]:
```suggestion
		Incorrect: A hallmark of problems featuring complete search with permutations/subsets is that $n$ is small, usually $\leq 20$. This means that recursion won't go particularly deep, and there won't be any issues with stack size. 
```

File: content/2_Bronze/Complete_Rec.mdx [hunk 2]
```diff
@@ -985,3 +990,324 @@ appear much less frequently then they once did.
...
+	    <Quiz.Answer>
+	      Recursion often runs into the issue of stack size, which can cause unexpected runtime errors
+	      <Quiz.Explanation>
+		A hallmark of problems featuring complete search with permutations/subsets is that $n$ is small, usually $\leq 20$. This means that recursion won't go particularly deep, and there won't be any issues with stack size. 
+	      </Quiz.Explanation>
+	    </Quiz.Answer>
+	    <Quiz.Answer correct>
+	      Iterative methods are easier to implement and are often faster and less buggy. 
+	      <Quiz.Explanation>
+		Having to code your own permutation/subset function can be inefficient in a timed contest environment. It's definitely worth learning iterative methods to save valuable time on debugging.
```
Comment (Aadit-Ambadkar) [inline content/2_Bronze/Complete_Rec.mdx:1062]:
```suggestion
		Correct: Having to code your own permutation/subset function can be inefficient in a timed contest environment. It's definitely worth learning iterative methods to save valuable time on debugging. 
```

File: content/2_Bronze/Complete_Rec.mdx [hunk 3]
```diff
@@ -985,3 +990,324 @@ appear much less frequently then they once did.
...
+	    <Quiz.Answer correct>
+	      Iterative methods are easier to implement and are often faster and less buggy. 
+	      <Quiz.Explanation>
+		Having to code your own permutation/subset function can be inefficient in a timed contest environment. It's definitely worth learning iterative methods to save valuable time on debugging. 
+	      </Quiz.Explanation>
+	    </Quiz.Answer>
+	    <Quiz.Answer>
+	      Recursion is so slow that it will often cause Time Limit Exceeded even for small arrays and should be avoided at all costs. 
+	      <Quiz.Explanation>
+		Although recursion is slowing than iterative methods, it's still fast enough to pass most problems
```
Comment (Aadit-Ambadkar) [inline content/2_Bronze/Complete_Rec.mdx:1068]:
```suggestion
		Incorrect: Although recursion is slowing than iterative methods, it's still fast enough to pass most problems
```

File: content/2_Bronze/Complete_Rec.mdx [hunk 4]
```diff
@@ -985,3 +990,324 @@ appear much less frequently then they once did.
...
+	      </Quiz.Explanation>
+	    </Quiz.Answer>
+	  </Quiz.Question>
+	</Quiz>    
+    </PySection>
+	
+    <JavaSection>
```
Comment (Aadit-Ambadkar) [inline content/2_Bronze/Complete_Rec.mdx:1075]: Repeat C++ Comments

File: content/2_Bronze/Complete_Rec.mdx [hunk 5]
```diff
@@ -985,3 +990,324 @@ appear much less frequently then they once did.
...
+	    <Quiz.Answer>
+	      Recursion is so slow that it will often cause Time Limit Exceeded even for small arrays and should be avoided at all costs. 
+	      <Quiz.Explanation>
+		Although recursion is slowing than iterative methods, it's still fast enough to pass most problems
+	      </Quiz.Explanation>
+	    </Quiz.Answer>
+	  </Quiz.Question>
+	</Quiz>    
+    </JavaSection>
+    <CPPSection>
```
Comment (Aadit-Ambadkar) [inline content/2_Bronze/Complete_Rec.mdx:1154]: Add Correct and Incorrect for each of these

**Review by Aadit-Ambadkar — CHANGES_REQUESTED** [review]

**Comment by Aadit-Ambadkar** [comment]
Add Correct and Incorrect to the beginning of each Answer Explanation.

**Comment by SansPapyrus683** [comment]
> > who the hell would learn arrays before vectors
> 
> Language

arpan don't say that i've seen you say the f word in the discord

**Review by SansPapyrus683 — APPROVED** [review]
should be good enough
also if we have to add Correct of Incorrect in front of each answer explanation
why not just automate it or something

**Comment by SansPapyrus683** [comment]
bruh

**Pull request closed without merging**